scripts/emit-feed-changelog-markdown.py

```diff
@@ -0,0 +1,86 @@
+#   the JSON
+
+from argparse import ArgumentParser, Namespace
+from json import loads
+
+# The start text of the FEED_CHANGELOG.md file
+FEED_CHANGELOG_START: str = \
+    '# Changes to NI Linux RT Package Feeds\n\n' \
+        + '## Description\n\n' \
+        + 'This documents the full changes to the core/ and extra/ feeds for each version ' \
+        + 'release.\n\n' \
+        + 'The changes are too long to keep in the [Changelog](/CHANGELOG.md) itself, ' \
+        + 'so they are kept here\n\n' \
+        + 'Generate them by:\n    1. adding a line to ./docs/feed-changelog.json: ' \
+        + '`./scripts/diff-pkg-changes.py <branch of old version> <old version> >>' \
+        + '<branch of new version> <new version> >> ./docs/feed-changelog.json`\n    ' \
+        + '2. updating the markdown with `./scripts/emit-feed-changelog-markdown.py ' \
+        + 'feed-changelog.json > feed-changelog.md`\n\n'
+
+def main(args: Namespace) -> int:
+    # Read the text in as string
+    json_str: str = ''
+    with open(args.input_file) as json_file:
+        json_str = json_file.read()
+
+    # Then split into lines of json
+    json_lines: list[str] = json_str.split('\n')
+    data: dict = {}
+    for line in json_lines:
+        if line == '':
+            continue
+        json_line: dict = loads(line)
+        for key, val in json_line.items():
+            data[key] = val
+
+    # Create a report from that info
+    report: str = FEED_CHANGELOG_START
+    for change_name, change_data in data.items():
+        report += \
+            '## Feed Changes from Version ' \
+                + change_data['old_vers'] + ' to Version ' + change_data['new_vers'] \
+                + '\n\n### Changes to core/\n\n* Added:\n'
+        for pkg in change_data['core']['added']:
+            if not pkg['name'].endswith('-lic'):
+                report += '    * ' + pkg['name'] + '\n'
+        report += '* Removed:\n'
+        for pkg in change_data['core']['removed']:
+            if not pkg['name'].endswith('-lic'):
+                report += '    * ' + pkg['name'] + '\n'
+        report += '\n### Changes to extra/\n\n* Added:\n'
+        for pkg in change_data['extra']['added']:
+            if not pkg['name'].endswith('-lic'):
+                report += '    * ' + pkg['name'] + '\n'
+        report += '* Removed:\n'
+        for pkg in change_data['extra']['removed']:
+            if not pkg['name'].endswith('-lic'):
+                report == '    * ' + pkg['name'] + '\n'
+        report += '\n'
+
+    print(report)
+
+    return 0
+
+if __name__ == '__main__':
+    # Add an argument for the JSON file
+    parser = ArgumentParser(
+        description = 'Convert JSON changelog data into a readable report'
+    )
+    parser.add_argument(
+        'input_file',
+        metavar = 'INPUT', type = str,
+        help = 'The JSON data file to read'
+    )
+    args = parser.parse_args()
```

Comment:
Just read the json list from stdin. Stdin is your friend. :)

```python3
for line in sys.stdin:
    json_line = json.loads(line)
...
```

Then you don't need the argparsing bits at all.